pkg/runner/healthcheck_common.go

```diff
@@ -0,0 +1,256 @@
+package runner
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"os/exec"
+	"reflect"
+
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/client"
+	"github.com/docker/go-connections/nat"
+	"github.com/docker/go-units"
+	"github.com/ipfs/testground/pkg/api"
+	"github.com/ipfs/testground/pkg/docker"
+
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+)
+
+type Checker func() bool
```

Comment:
The check can error out, which does not denote that the check failed, but that it was inoperative altogether. I think this function needs to return `(bool, error)`.